src/renderer/components/+extensions/extensions.tsx

```diff
@@ -32,27 +54,238 @@ export class Extensions extends React.Component {
     return extensionManager.localFolderPath;
   }
 
+  getExtensionDestFolder(name: string) {
+    return path.join(this.extensionsPath, sanitizeExtensionName(name));
+  }
+
+  installFromSelectFileDialog = async () => {
+    const { dialog, BrowserWindow, app } = remote;
+    const { canceled, filePaths } = await dialog.showOpenDialog(BrowserWindow.getFocusedWindow(), {
+      defaultPath: app.getPath("downloads"),
+      properties: ["openFile", "multiSelections"],
+      message: _i18n._(t`Select extensions to install (formats: ${this.supportedFormats.join(", ")}), `),
+      buttonLabel: _i18n._(t`Use configuration`),
+      filters: [
+        { name: "tarball", extensions: this.supportedFormats }
+      ]
+    });
+    if (!canceled && filePaths.length) {
+      this.requestInstall(
+        filePaths.map(filePath => ({
+          fileName: path.basename(filePath),
+          filePath: filePath,
+        }))
+      );
+    }
+  };
+
+  installExtensions = () => {
+    if (this.downloadUrl) {
+      this.installFromNpmOrUrl(this.downloadUrl);
+      this.downloadUrl = "";
+    } else {
+      this.installFromSelectFileDialog();
+    }
+  };
+
+  installFromNpmOrUrl = async (url = this.downloadUrl) => {
+    if (!InputValidators.isUrl.validate(url)) {
+      const npmPackageName = url;
+      url = extensionManager.getNpmPackageTarballUrl(npmPackageName);
```

Comment:
I'd avoid reassigning to function parameters. We should also configure this with eslint (no-param-reassign).

Reply:
Why? Do you have clear explanation why it's bad? :)

Reply:
Answered in the other comment.